src/rust/ide/src/controller/project.rs

```diff
@@ -153,14 +174,58 @@ impl Handle {
                     }
                 }
                 Event::Closed => {
-                    error!(logger,"Lost binary data connection!");
+                    error!(logger,"Lost binary connection with the Language Server!");
+                    // TODO [wmu]
+                    //  The problem should be reported to the user and the connection should be
+                    //  reestablished, see https://github.com/luna/ide/issues/145
+                }
+                Event::Error(error) => {
+                    error!(logger,"Error emitted by the binary data connection: {error}.");
+                }
+            }
+            futures::future::ready(())
+        }
+    }
+
+    /// Returns a handling function capable of processing updates from the json-rpc protocol.
+    /// Such function will be then typically used to process events stream from the json-rpc
+    /// connection handler.
+    pub fn json_event_handler
+    (&self) -> impl Fn(enso_protocol::language_server::Event) -> futures::future::Ready<()> {
+    // TODO [mwu]
+    //  This handler for JSON-RPC notifications is very similar to the function above that handles
+    //  binary protocol notifications. However, it is not practical to generalize them, as the
+    //  underlying RPC handlers and their types are separate.
+    //  This generalization should be reconsidered once the old JSON-RPC handler is phased out.
```

Comment:
We have already an issue for unifying handlers, don't we?
If we have, its description should also contain note to refactor this functions according to this TODO.